### PR TITLE
Contract size

### PR DESCRIPTION
**Pull request by grandizzy**
Reducing ERC721Pool size from 29643 to 26761 bytes and ERC20Pool from 25122 to 24784 bytes
I noticed that if replacing Maths.WAD / RAY with inline 10 ** 18 / 10 ** 27 in these contracts drastically reduce the size but will keep this as a last resort in case moving to custom errors is still not enough (from my quick tests it is enough)

Changes:
- keep same contract structure / functions for NFT collateral pool. Remove `multiple` and make multiple default, callers could pass an array with one element if adding / removing / claiming single token
- iterate only once through token ids, remove internal functions to check token ids
- ERC271 initialize subset to reuse initialize function
- Refactor estimate price function, remove view from buckets  and keep view only at pool contract level
- Remove Maths.ONE_WAD and use Maths.WAD which is equivalent

### CODE REVIEW

File: src/PositionManager.sol [hunk 1]
```diff
@@ -55,12 +55,6 @@ contract PositionManager is IPositionManager, Multicall, PositionNFT, PermitERC2
 
         pool.removeQuoteToken(params_.recipient, quoteTokenToRemove, params_.price);
 
-        // enable lenders to remove quote token from a bucket that no debt is added to
-        if (collateralToRemove != 0) {
-            // claim any unencumbered collateral accrued to the price bucket
-            pool.claimCollateral(params_.recipient, Maths.rayToWad(collateralToRemove), params_.price);
```
Comment (MikeHathaway) [inline src/PositionManager.sol:61]: Why was this removed?

File: src/base/Buckets.sol [hunk 2]
```diff
@@ -191,7 +191,7 @@ abstract contract Buckets is IBuckets {
         }
     }
 
-    function _claimMultipleNFTCollateralFromBucket(uint256 price_, uint256[] memory tokenIds_, uint256 lpBalance_) internal returns (uint256 lpRedemption_) {
+    function _claimNFTCollateralFromBucket(uint256 price_, uint256[] memory tokenIds_, uint256 lpBalance_) internal returns (uint256 lpRedemption_) {
```
Comment (MikeHathaway) [inline src/base/Buckets.sol:158]: The other `_claimNFTCollateralFromBucket` method right above can be removed as well

**Comment by MikeHathaway** [comment]
Generally LGTM. One additional method can be removed, and has a question about a removal from `PositionManager` which I think is s till needed.

Passing an array for single tokens will slightly increase gas cost, but it should be relatively minimal and is probably one of the better ways to get contract size down.

File: src/base/Buckets.sol (hunk 2 above)
Comment (grandizzy) [reply src/base/Buckets.sol:158]: ah, yep, removed with https://github.com/ajna-finance/contracts/pull/149/commits/8074a633e612a7106d9ef5f89d85e90cbe25961e

File: src/PositionManager.sol (hunk 1 above)
Comment (grandizzy) [reply src/PositionManager.sol:61]: That's because the NFT pool contract only allows now an array of tokenIds `function claimCollateral(address recipient_, uint256[] calldata tokenIds_, uint256 price_)`. Maybe we need here an `decreaseNFTLiquidity` function?

Comment (MikeHathaway) [reply src/PositionManager.sol:61]: Gotcha, yeah I can add that as part of the positionManager upgrades following finishing with `repay()`

**Review by MikeHathaway — APPROVED** [review]
LGTM

File: src/PositionManager.sol (hunk 1 above)
Comment (grandizzy) [reply src/PositionManager.sol:61]: on a second thought, do we really need claim collateral on decrease liquidity function?

Comment (MikeHathaway) [reply src/PositionManager.sol:61]: I think so - this method was intended to decrease the amount of LP tokens wrapped in the position which is how the claim on collateral would be represented
